core/lib/web3_decl/src/error.rs

```diff
@@ -42,3 +43,65 @@ pub enum Web3Error {
     #[error("Tree API is not available")]
     TreeApiUnavailable,
 }
+
+use std::{collections::HashMap, error, fmt, fmt::Debug};
+
+#[derive(Debug)]
+pub struct RpcErrorWithDetails {
+    pub inner_error: RpcError,
+    pub method: String,
+    pub args: HashMap<String, String>,
+}
+impl error::Error for RpcErrorWithDetails {}
```

Comment:
You should override [`source()`](https://doc.rust-lang.org/std/error/trait.Error.html#method.source) method to point at the original `RpcError`.